tokio/src/io/util/copy.rs

```diff
@@ -27,6 +27,46 @@ impl CopyBuffer {
         }
     }
 
+    fn poll_fill_buf<R>(
+        &mut self,
+        cx: &mut Context<'_>,
+        reader: Pin<&mut R>,
+    ) -> Poll<io::Result<()>>
+    where
+        R: AsyncRead + ?Sized,
+    {
+        let me = &mut *self;
+        let mut buf = ReadBuf::new(&mut me.buf[me.cap..]);
+
+        let res = reader.poll_read(cx, &mut buf);
+        me.cap += buf.filled().len();
```

Comment:
This could also be written like this.
```suggestion
        let mut buf = ReadBuf::new(&mut me.buf);
        buf.set_filled(me.cap);
        let res = reader.poll_read(cx, &mut buf);
        me.cap = buf.filled().len();
```

Reply:
```suggestion
        let mut buf = ReadBuf::new(&mut me.buf[me.cap..]);
        buf.set_filled(me.cap);
        let res = reader.poll_read(cx, &mut buf);
        let filled_len = buf.filled.len();
        self.read_done = me.cap == filled_len;
        me.cap = filled_len;
```

Reply:
I've taken your suggestion, since I believe it makes it easier to micro-optimize with `ReadBuf::uninit` safely in future, if that shows up in a profile.